bpf/trace_common.h

```diff
@@ -209,14 +223,22 @@ static __always_inline void get_or_create_trace_info(http_connection_metadata_t
             //bpf_dbg_printk("Looking up existing trace for connection");
             //dbg_print_http_connection_info(conn);
 
-            tp_info_pid_t *existing_tp = trace_info_for_connection(conn);
-
-            if (correlated_requests(tp_p, existing_tp)) {
+            tp_info_pid_t *existing_tp = bpf_map_lookup_elem(&incoming_trace_map, conn);
```

Comment:
For server requests, we first look for TCP info and then we fall back to black-box info.